appsec/src/extension/ddappsec.c

```diff
@@ -438,31 +343,29 @@ __thread void *unspecnull TSRMLS_CACHE = NULL;
 
 static void _check_enabled()
 {
-    if (!get_global_DD_APPSEC_TESTING() &&
-        (!dd_trace_enabled() || strcmp(sapi_module.name, "cli") == 0 ||
-            (sapi_module.phpinfo_as_text != 0))) {
-        DDAPPSEC_G(enabled_by_configuration) = DISABLED;
-    } else if (!dd_is_config_using_default(DDAPPSEC_CONFIG_DD_APPSEC_ENABLED)) {
-        DDAPPSEC_G(enabled_by_configuration) =
-            get_DD_APPSEC_ENABLED() ? ENABLED : DISABLED;
+    if ((!get_global_DD_APPSEC_TESTING() && !dd_trace_enabled()) ||
+        (strcmp(sapi_module.name, "cli") != 0 && sapi_module.phpinfo_as_text)) {
+        DDAPPSEC_G(enabled) = APPSEC_FULLY_DISABLED;
+        DDAPPSEC_G(active) = false;
+        DDAPPSEC_G(to_be_configured) = false;
+    } else if (!dd_cfg_enable_via_remcfg()) {
+        DDAPPSEC_G(enabled) = get_DD_APPSEC_ENABLED() ? APPSEC_FULLY_ENABLED
+                                                      : APPSEC_FULLY_DISABLED;
+        DDAPPSEC_G(active) = get_DD_APPSEC_ENABLED() ? true : false;
+        DDAPPSEC_G(to_be_configured) = false;
     } else {
-        DDAPPSEC_G(enabled_by_configuration) = NOT_CONFIGURED;
+        DDAPPSEC_G(enabled) = APPSEC_ENABLED_VIA_REMCFG;
```

Comment:
Before and now(I believe) on every request, the extension was doing a `config_sync` when the app was not disabled by config and not enabled by RC. This was done on purpose. Why do you think it's a bug?

Reply:
It's a bug because that's not what I defined in the protocol RFC. `config_sync` was only intended as a mechanism to enable a disabled extension through remote config, changing the state was meant to be done by responding to a `request_init` with a `config_features`.

IIRC we discussed at length the new requirement of being able to have multiple responses to a single message so that we could update the extension configuration during a call to `request_init`, while still being enabled (e.g. if we had the ability to change the extension's configuration through RC, which we currently don't).

In saying that, I guess it's fair to assume that there are other pieces missing from the current implementation.

Reply:
> config_sync was only intended as a mechanism to enable a disabled extension through remote config
This is what it does, so on every request after the extension has been disabled by RC, it does a `config_sync` to check if it has to be enabled or not

Reply:
No, it currently sends it on every request even if it's enabled.